### DISCUSSION TITLE
Custom 404 not-found page not working in Vercel

### DISCUSSION THREAD

**Discussion by ShueiYang** [vercel/next.js#54141] [Help] [answered]
### Summary

Hello,
i have a strange issue with Nextjs 13 app dir, I place a not-found.tsx file in the app folder and it is working in the localhost, and also in production build in localhost, but as i push to production at vercel i will still see the default 404 page from Vercel, is there something I miss to set up in Vercel or there is something else ?

Regards

**Comment by icyJoseph** [comment]
Next.js version? The one I have deployed is working correctly though.

**Reply by ShueiYang** [reply]
Well the last version 13.4.12, now that I check again on another personal project I made before I have the same  Nextjs version since i update  and the custom 404 not-found.ts is  WORKING.... Now i am confuse how come the second project didn't work although i am using more or less the same code and name.... there must something else on my code surely but I have no idea how to debug this thing : /

**Comment by ShueiYang** [comment, marked as answer]
Ok so I try to host on another provider in Docker  and the custom 404 page is working, looks like it didn't work only when I deploy in Vercel in 13.4.12. And once I update into 13.4.19  now the custom 404 is finally working in Vercel.

**Reply by DongeunPaeng** [reply]
Maybe have you deleted the original not-found.tsx file and remade the file? I did so and experienced the exact same problem as yours.

**Comment by linkb15** [comment]
I am facing this issue in Next 14.0.1 in the Vercel with combination of `pages/404.tsx` and `app/not-found.tsx`

**Reply by zahid726** [reply]
https://dev.to/stanlisberg/resolving-the-vercel-404-page-not-found-error-after-page-refresh-9b9#:~:text=To%20rectify%20the%20404%20Not,root%20directory%20of%20your%20project.
